src/WorldSelector.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React from 'react';
+import type { WorldName } from './types';
+import type { IntlShape } from 'react-intl';
+import { injectIntl } from 'react-intl';
+
+import {ReactComponent as HeaderIcon} from './svg/SelectWorld.svg'
+import {ReactComponent as RabbitIcon} from './svg/Rabbit.svg';
+import {ReactComponent as RobotIcon} from './svg/Robot.svg';
+import { ReactComponent as SpaceShipIcon } from './svg/SpaceShip.svg';
+
+import './WorldSelector.scss';
+
+type WorldSelectorProps = {
+    intl: IntlShape,
+    onSelect: (value: WorldName) => void
+};
+
+type WorldSelectorState = {
+    world: WorldName
+}
+
+class WorldSelector extends React.Component<WorldSelectorProps, WorldSelectorState> {
+    constructor (props) {
+        super(props);
+        this.state = {
+            world: 'default'
+        }
+    }
+
+    componentDidUpdate(prevProps: WorldSelectorProps, prevState: WorldSelectorState) {
+        if (this.state.world !== prevState.world) {
+            this.props.onSelect(this.state.world);
+        }
+    }
+
+    handleCharacterClick = (event: Event, worldName: WorldName) => {
+        event.preventDefault();
+        this.setState({ world: worldName });
+    }
+
+    handleCharacterKeyDown = (event: KeyboardEvent, worldName: WorldName) => {
+        const toggleKeys = [' ', 'Enter'];
+
+        if (toggleKeys.indexOf(event.key) !== -1) {
+            event.preventDefault();
+            this.setState({ world: worldName });
+        }
+    }
+
+    render() {
+        return (
+            <div className='WorldSelector'>
+                <HeaderIcon
+                    aria-label={this.props.intl.formatMessage({id:'WorldSelector.world'})}
+                    className="HeaderIcon"
+                />
+                <RobotIcon
```

Comment:
Let's wrap this icon with a button, and move all the properties in this SVG to it. Also, it would be nice to make this as a toggle button by using aria-pressed property, so users know if the character is selected or not. We also need value property, that's assigned to each character name, in this case, 'default'. Note that we need to use world property from App, with changes to removing state from this component. 

`<button className= {"WorldIcon" + (this.props.world === 'default' ? " WorldIcon--selected" : "") } aria-label={this.props.intl.formatMessage({id:'WorldSelector.world.default}} value='default' onClick={this.handleCharacterClick} onKeyDown={this.handleCharacterKeyDown}> 

`

Reply:
Good idea, and done.